### PR TITLE
Add INT(16) data type

### PR DESCRIPTION
**Pull request by yujong-lee**
I am adding the `INT(16)` data type to the core and encountered one problem.

: [This commented test](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/test-suite/src/data_type/int16.rs#L62-L65) fails.

스크린샷 2022-07-10 오후 10 18 23

Seems like it goes to [here](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/core/src/data/value/binary_op/i16.rs#L110-L114) rather than [here](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/core/src/data/value/binary_op/i16.rs#L63-L73) but I don't know why.

Can anyone guide me to figure this out?

### CODE REVIEW

**Comment by ever0de** [comment]
> I am adding the `INT(16)` data type to the core and encountered one problem.
> 
> : [This commented test](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/test-suite/src/data_type/int16.rs#L62-L65) fails.
> 
> 스크린샷 2022-07-10 오후 10 18 23
> 
> Seems like it goes to [here](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/core/src/data/value/binary_op/i16.rs#L110-L114) rather than [here](https://github.com/yujong-lee/gluesql/blob/b2f28018cb9ee7274fc9c5f076f91aae0ed40aa0/core/src/data/value/binary_op/i16.rs#L63-L73) but I don't know why.
> 
> Can anyone guide me to figure this out?

There is one implementation missing.
Climbing through the error, check the `add` method of the `Value` struct implementation.
The `I16` implementation is missing.
` (I16(a), b) => a.try_add(b),`

**Comment by yujong-lee** [comment]
@ever0de Thank you so much.

File: core/src/data/interval/primitive.rs [hunk 1]
```diff
@@ -109,6 +128,17 @@ impl Div<i8> for Interval {
...
 }
 
+impl Div<i16> for Interval {
+    type Output = Self;
+
+    fn div(self, rhs: i16) -> Self {
+        match self {
+            Interval::Month(v) => Interval::Month(v / rhs as i32),
+            Interval::Microsecond(v) => Interval::Microsecond(v / rhs as i64),
+        }
```
Comment (MRGRAVITY817) [inline core/src/data/interval/primitive.rs:138]: This `match self` part appears similarly on every `impl Div<_> for Interval { ... }`s.
Would be nice to refactor them with a helper function in near future.

**Review by MRGRAVITY817 — APPROVED** [review]
LGTM 👍

**Review by panarch — APPROVED** [review]
Awesome!
Looks all nice, thanks a lot! 👍 👍 👍

**Pull request merged**